### DISCUSSION TITLE
Master Thesis on JSBSim integrated with hardware

### DISCUSSION THREAD

**Discussion by sthelia** [JSBSim-Team/jsbsim#492] [Show and tell]
Hi everyone! 

As mentioned in some of my previous questions I have worked on implementing JSBSim with a test rig, something that resulted in the Matlab/Simulink integration PR. For those interested, I'd like to share the thesis I wrote on it. 

The thesis can be found [here!](http://kth.diva-portal.org/smash/record.jsf?aq2=%5B%5B%5D%5D&c=30&af=%5B%5D&searchType=LIST_LATEST&sortOrder2=title_sort_asc&query=&language=en&pid=diva2%3A1583510&aq=%5B%5B%5D%5D&sf=all&aqe=%5B%5D&sortOrder=author_sort_asc&onlyFullText=false&noOfRows=50&dswid=-476)

**Comment by seanmcleod** [comment]
Thanks for the link to your thesis @sthelia.

Couple of questions after reading it. Was there a particular reason you used JSBSim as opposed to using the Aerospace Blockset of Matlab? 

Can you provide a link to your GAM/ADMIRE JSBSim model?

**Reply by sthelia** [reply]
Yes, the reason for using JSBSim was first that we wanted to investigate the usability of an open-source simulator. The original plan was not to use the Simulink integration, but instead to either communicate through TCP/UDP or maybe create a C program to run it. However, the real-time computer (Speedgoat) that was used to control the Iron Bird test rig is based on Simulink, so Mathworks advised us to use S-functions. 
I haven't looked into the Aerospace Blockset, but another thing we really liked with JSBSim was how easy it was to implement new aircraft into simulations, also something that the company that supported the thesis was interested in. 

Unfortunately, I'm not allowed to share the GAM model online. The data should be public, but I can't find it when I search for it. In this paper attached, they provide an email address that might work if someone would like access to the data, but I'm not sure. (admire @ foi.se)
[FOIR1624.pdf](https://github.com/JSBSim-Team/jsbsim/files/6966536/FOIR1624.pdf)

**Reply by seanmcleod** [reply]
Ah, okay. When I read the thesis and saw that you were integrating with Matlab/Simulink to integrate with the IronBird I wondered why you hadn't just used Matlab for the aerodynamic modelling as well then.

In terms of the GAM/ADMIRE data I have a copy of the papers and the data, but haven't had the time to convert the data into a JSBSim format yet, so I thought of using your JSBSim model as a starting point. 

The ADMIRE paper mentions:

> A secondary goal was to provide the research community with an unlimited model that could freely be distributed.

And the data was available for download for many years, but packaged for use with Matlab/Simulink. But at some point the download link disappeared, but without any explanation or notice that the 'freely distributed' allowance had been revoked.

**Reply by sthelia** [reply]
> Ah, okay. When I read the thesis and saw that you were integrating with Matlab/Simulink to integrate with the IronBird I wondered why you hadn't just used Matlab for the aerodynamic modelling as well then.

Good point, it would for sure have worked if the only goal was to integrate it with the specific hardware. 

> In terms of the GAM/ADMIRE data I have a copy of the papers and the data, but haven't had the time to convert the data into a JSBSim format yet, so I thought of using your JSBSim model as a starting point.

Ah, I see. Yes, it is a tedious process to convert the data. I used 3D-tables in JSBSim and used Matlab interpolation to convert the data to something JSBSim would understand. Sorry I can't be of more help there!

**Comment by rubendevalois** [comment]
Nice work Tilda, congratulations and thanks for sharing !!!

rgrds, - Ruben

**Reply by sthelia** [reply]
Thank you!
